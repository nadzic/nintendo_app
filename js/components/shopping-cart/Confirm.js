import React, { Component } from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { get as safeGet } from 'lodash';
import { COLORS } from 'nintendoapp/js/constants/colors';
import { calculateTotal } from 'nintendoapp/js/utils/calculation';


/* Config/Constants
============================================================================= */

/* eslint-disable global-require */

const FIELDS_WIDTH = '88%';
const BUTTON_HEIGHT = 50;

type Props = {
  navigation: any,
  ui: any
};

class Confirm extends Component<Props> {
  // TODO: disconnect from store
  goToNextScreen() {
    const {
      navigation,
    } = this.props;

    navigation.navigate('Payment');
  }

  render() {
    const products = safeGet(this.props, 'products', []);

    const numProducts = products.length;
    const total = calculateTotal(products);
    return (
      <View style={[styles.container, numProducts < 1 && styles.disabled]}>
        <View style={styles.totalSection}>
          <Text style={styles.totalText}>Total : </Text>
          <Text style={styles.valueText}>${total}</Text>
        </View>
        <TouchableOpacity
          style={styles.confirmButton}
          onPress={() => this.goToNextScreen()}
          disabled={numProducts < 1}
        >
          <Text style={styles.confirmText}>Confirm and Pay</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: COLORS.LIGHTGREY3,
    position: 'absolute',
    width: '100%',
    bottom: 0,
    borderTopColor: COLORS.LIGHTGREY,
    borderTopWidth: 0.5,
    paddingTop: 7,
  },
  disabled: {
    opacity: 0.3,
  },
  totalSection: {
    display: 'flex',
    flexDirection: 'row',
  },
  totalText: {
    color: COLORS.BLACK,
    fontSize: 18,
    marginBottom: 8,
    fontFamily: 'BrandonText-Bold',
  },
  valueText: {
    color: COLORS.GREEN,
    fontSize: 18,
    fontFamily: 'BrandonText-Bold',
  },
  confirmButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.GREEN,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginBottom: 15,
  },
  confirmText: {
    fontFamily: 'BrandonText-Medium',
    color: COLORS.WHITE,
    fontSize: 20,
  },
});

export default Confirm;
