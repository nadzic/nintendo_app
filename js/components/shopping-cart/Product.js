import React, { Component } from 'react';
import { Text, Image, View, StyleSheet, TouchableOpacity } from 'react-native';
import { COLORS } from 'nintendoapp/js/constants/colors';

/* eslint-disable global-require */

class Product extends Component {
  render() {
    const { product, onPress, index } = this.props;

    if (!product) {
      return null;
    }

    return (
      <View style={styles.container}>
        <Image
          source={{ uri: product.imageUrl }}
          style={styles.productImage}
          resizeMode="contain"
        />
        <View style={styles.textSection}>
          <Text style={styles.shortText}>{product.short}</Text>
          <Text style={styles.nameText}>{product.name}</Text>
          <Text style={styles.flavorText}>{product.flavor}</Text>
          <Text style={styles.productText}>${product.price} x 1</Text>
        </View>
        <TouchableOpacity
          style={styles.deleteImg}
          onPress={onPress}
          testID={`Delete Product ${index} Shopping Cart`}
        >
          <Image
            source={require('nintendoapp/js/components/shopping-cart/img/delete.png')}
          />
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.WHITE,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: 15,
    borderBottomColor: COLORS.LIGHTGREY,
    borderBottomWidth: 0.5,
  },
  textSection: {
    paddingLeft: 20,
    width: '100%',
  },
  productImage: {
    width: 67,
    height: 67,
  },
  deleteImg: {
    position: 'absolute',
    display: 'flex',
    alignItems: 'flex-end',
    right: 15,
  },
  shortText: {
    fontFamily: 'BrandonText-Medium',
    fontSize: 15,
    color: COLORS.DARKGREY,
    lineHeight: 20,
  },
  nameText: {
    fontFamily: 'BrandonText-Bold',
    fontSize: 16,
    color: COLORS.BLACK,
    lineHeight: 20,
    width: '70%',
  },
  flavorText: {
    fontSize: 15,
    fontFamily: 'BrandonText-Medium',
    fontWeight: '500',
    color: COLORS.DARKGREY,
    lineHeight: 20,
  },
  productText: {
    fontFamily: 'BrandonText-Medium',
    fontSize: 14,
    lineHeight: 20,
  },
});

export default Product;
