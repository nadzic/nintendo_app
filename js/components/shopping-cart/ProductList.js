import React, { Component } from 'react';
import { View, StyleSheet, Text, ScrollView } from 'react-native';
import Product from 'nintendoapp/js/components/shopping-cart/Product';
import { COLORS } from 'nintendoapp/js/constants/colors';
import { getIndexInUpcs } from 'nintendoapp/js/utils/calculation';

/* eslint-disable global-require */

class ProductList extends Component {
  onDeleteProductPress = (product) => {
    const {
      quantities,
      upcs,
      deleteProductFromCart,
      deleteQuantityFromCart,
      deleteVariantFromCart,
      decreaseQuantityInCart,
      deleteUpcFromCart,
    } = this.props;
    const index = getIndexInUpcs(product.upcCode, upcs);
    if (quantities[index] < 2) {
      deleteProductFromCart(product.id);
      deleteQuantityFromCart(index);
      deleteVariantFromCart(index);
      deleteUpcFromCart(index);
    } else {
      deleteProductFromCart(product.id);
      decreaseQuantityInCart(index);
    }
  }

  render() {
    const {
      products,
    } = this.props;

    if (!products) {
      return null;
    }

    return (
      <ScrollView
        style={styles.container}
        testID='Scroll View Shopping Cart'
      >
        {products.length > 0 ? products.map((product, index) => (
          <Product
            product={product}
            key={product.id}
            onPress={() => this.onDeleteProductPress(product)}
            index={index}
          />
        )) :
        <View style={styles.emptyCard}>
          <Text style={styles.emptyCardText}>{`Your cart is empty.\nPlease scan to add items.`}</Text>
        </View>}
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: COLORS.LIGHTGREY2,
    marginBottom: 105.5,
  },
  emptyCard: {
    marginTop: 120,
    display: 'flex',
    alignItems: 'center',
  },
  emptyCardText: {
    fontFamily: 'BrandonText-Medium',
    fontSize: 21,
    textAlign: 'center',
    color: COLORS.DARKGREY,
  },
});

export default ProductList;
