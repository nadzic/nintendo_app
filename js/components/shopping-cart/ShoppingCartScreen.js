import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { View, StyleSheet, TouchableOpacity, Image } from 'react-native';
import { get as safeGet } from 'lodash';
import { COLORS } from 'nintendoapp/js/constants/colors';
import ProductList from 'nintendoapp/js/components/shopping-cart/ProductList';
import Confirm from 'nintendoapp/js/components/shopping-cart/Confirm';
import {
  deleteProductFromCart,
  deleteQuantityFromCart,
  deleteVariantFromCart,
  decreaseQuantityInCart,
  deleteUpcFromCart,
} from 'nintendoapp/js/actions/cartActions';
import { getCreditCard } from 'nintendoapp/js/actions/paymentActions';
import Loader from 'nintendoapp/js/components/common/Loader';
import NavigationService from 'nintendoapp/js/utils/navigation';

/* eslint-disable global-require */

type Props = {
  products: Array,
  quantities: Array,
  navigation: any,
};

export class ShoppingCartScreen extends Component<Props> {
  static navigationOptions = {
    title: 'Shopping cart',
    headerTitleStyle: {
      fontSize: 20,
      fontWeight: 'normal',
      fontFamily: 'BrandonText-Medium',
    },
    headerStyle: {
      backgroundColor: COLORS.GREEN,
    },
    headerTintColor: COLORS.WHITE,
    headerLeft: (
      <TouchableOpacity
        style={{ marginLeft: 15 }}
        onPress={() => NavigationService.resetState('ScanProduct')}
      >
        <Image
          style={{ width: 30, height: 30 }}
          source={require('nintendoapp/js/components/shopping-cart/img/back.png')}
        />
      </TouchableOpacity>),
  }

  render() {
    const {
      navigation,
      products,
      quantities,
      upcs,
      user,
      loading,
      deleteProductFromCart,
      deleteQuantityFromCart,
      deleteVariantFromCart,
      decreaseQuantityInCart,
      deleteUpcFromCart,
      getCreditCard,
    } = this.props;
    const auth_token = safeGet(user, 'auth_token', '');
    return (
      <View style={styles.container}>
        <ProductList
          products={products}
          quantities={quantities}
          upcs={upcs}
          deleteProductFromCart={deleteProductFromCart}
          deleteQuantityFromCart={deleteQuantityFromCart}
          deleteVariantFromCart={deleteVariantFromCart}
          decreaseQuantityInCart={decreaseQuantityInCart}
          deleteUpcFromCart={deleteUpcFromCart}
        />
        <Confirm
          products={products}
          quantities={quantities}
          navigation={navigation}
          getCreditCard={getCreditCard}
          authToken={auth_token}
        />
        {loading && <Loader />}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.LIGHTGREY2,
    height: '100%',
  },
});

const mapStateToProps = (state) =>
  ({
    user: state.auth.user,
    loading: state.loading,
    products: state.cart.products,
    quantities: state.cart.quantities,
    upcs: state.cart.upcs,
  });

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    deleteProductFromCart,
    deleteVariantFromCart,
    deleteQuantityFromCart,
    decreaseQuantityInCart,
    deleteUpcFromCart,
    getCreditCard,
  }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ShoppingCartScreen);
