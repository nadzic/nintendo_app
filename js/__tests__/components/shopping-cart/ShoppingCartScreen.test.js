import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ShoppingCartScreen } from 'nintendoapp/js/components/shopping-cart/ShoppingCartScreen';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
  };

  const wrapper = shallow(<ShoppingCartScreen {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('ShoppingCartScreen component', () => {
  it('should renders correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
