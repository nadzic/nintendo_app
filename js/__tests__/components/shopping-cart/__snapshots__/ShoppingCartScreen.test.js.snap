// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCartScreen component should renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "height": "100%",
    }
  }
>
  <ProductList />
  <Confirm
    authToken=""
  />
</View>
`;
